### PR TITLE
Support passphrases in keygen

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
Solana keygen cli tool doesn't support passphrases for BIP39 key generation. The BIP39 standard allows users to use an optional passphrase to increase the security of key recovery.

#### Summary of Changes
- `solana-keygen new` and `solana-keygen recover` now prompt for a passphrase by default
- Add --no-passphrase (-n) flag to `solana-keygen new` to skip passphrase prompt (for scripts)
- Add --skip-seed-phrase-validation flag to `solana-keygen recover` to recover a seed phrase which uses a non-standard or non-English word list
- Update scripts to skip passphrase prompts and to not display the mnemonic phrase

`solana-keygen new -so keypair.json`:
```
Generating a new keypair
For added security, enter a passphrase (empty for no passphrase):
Wrote new keypair to keypair.json
```

`solana-keygen recover`:
```
[recover] seed phrase: 
[recover] If this seed phrase has an associated passphrase, enter it now. Otherwise, press ENTER to continue: 
```

Fixes https://github.com/solana-labs/solana/issues/7088

### CODE REVIEW

File: keygen/src/keygen.rs [hunk 1]
```diff
@@ -72,11 +73,17 @@ fn main() -> Result<(), Box<dyn error::Error>> {
                         .long("force")
                         .help("Overwrite the output file if it exists"),
                 )
+                .arg(
+                    Arg::with_name("no_passphrase")
+                        .short("n")
```
Comment (t-nelson) [inline keygen/src/keygen.rs]: Actually... I changed my mind 🙃 

Let's drop the short opt here.  Make it a little painful to do it wrong

Comment (jstarry) [reply keygen/src/keygen.rs]: "wrong" 😝

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

File: keygen/src/keygen.rs (hunk 1 above)
Comment (mvines) [reply keygen/src/keygen.rs]: 👍 shortopts are the worst

**Pull request merged**